Assets/Scripts/Battlefield/CharacterFactory.cs

```diff
@@ -66,6 +66,7 @@ void Awake()
                 //brain
                 brain.tileIndictor = indicator;
                 brain.startCoordinates = new Vector2(numUnits, 5);
+                brain.tiles = grid.tiles;
```

Comment:
instead of holding the TIles[,] array. Take hold of the gridHolder itself since that is a much more robust object than the Tiles array.